main/background.ts

```diff
@@ -54,14 +55,36 @@ ipcMain.on("message", async (event, arg) => {
   event.reply("message", `${arg} World!`);
 });
 
+// TODO make generic function for creating pub+sub endpoints
+// TODO make class for subscription management
+let cachedMetrics: SetMetricsStateActionPayload | null = null;
+let cachedMetricsStr = "";
+// TODO list of webContents
+// let cachedMetricsSubList = [];
+let cachedMetricsIsSubActive = false;
+let cachedMetricsTimeout: NodeJS.Timeout | null = null;
```

Comment:
I'm a fan of this eslint rule https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-null.md in these sorts of situation, because it's less code to just write

```ts
let cachedMetricsTimeout: NodeJS.Timeout;
```

without assignment, it defaults to `NodeJS.Timeout | undefined`, null I feel should be used only when it's not a viable alternative to undefined, that's just opionated reply and you don't have to take this comment seriously

Reply:
I need explicit null because I need clear this timeout on ping

Reply:
but you can assign undefined to it?